server/src/inmantals/server.py

```diff
@@ -64,9 +65,149 @@ def __init__(self, message: str) -> None:
+        If this folder holds an inmanta project, this returns the path to this project.
+        If this folder holds an inmanta module, this returns the path of the temporary
+        inmanta project used to load this module.
+        """
+        if not self.inmanta_project_dir:
+            return None
+        if isinstance(self.inmanta_project_dir, str):
+            return self.inmanta_project_dir
+        else:
+            return self.inmanta_project_dir.name
+
+    def get_setting(self, key: str, default: Optional[str] = None) -> Optional[str]:
+        """
+        Get an inmanta extension setting for this folder
+        """
+        if key == "compiler_venv_path":  # support for versions of core that require a compiler venv.
+            return os.path.join(self.get_folder_path(), ".env-ls-compiler")
+        try:
+            # Look for some custom settings for this folder
+            with open(os.path.join(self.get_folder_path(), ".vscode", "settings.json")) as settings_file:
+                settings: dict[str, str] = json.load(settings_file)
+                return settings.get(key, default)
+        except FileNotFoundError:
+            return default
+
+    @classmethod
+    def unpack_workspaces(cls, workspace_folders: Sequence[object], ls_handler: "InmantaLSHandler") -> Dict[str, "Folder"]:
+        return {folder["uri"]: Folder(URI(folder["uri"]), folder["name"], ls_handler) for folder in workspace_folders}
+
+    def cleanup(self) -> None:
+        logger.info(f"calling cleanup for {self}")
+        if self.inmanta_project_dir:
+            self.handler.remove_folder(str(self.folder_uri))
+            self.inmanta_project_dir.cleanup()
+
+    def get_folder_path(self) -> str:
+        """
+        Convert the folders's uri into a regular path
+        (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#uri)
+        """
+        return self.folder_uri.path
+
+    def create_tmp_project(self) -> str:
+        """
+        When working on a module, an inmanta project is created in a temporary directory.
+        """
+        self.inmanta_project_dir = tempfile.TemporaryDirectory()
+        logger.debug(f"Temporary project created at {self.inmanta_project_dir.name}.")
+
+        os.mkdir(os.path.join(self.inmanta_project_dir.name, "libs"))
+
+        install_mode = module.InstallMode.master
+
+        v2_metadata_file: str = os.path.join(self.get_folder_path(), module.ModuleV2.MODULE_FILE)
+        v1_metadata_file: str = os.path.join(self.get_folder_path(), module.ModuleV1.MODULE_FILE)
+
+        module_name: Optional[str] = None
+        libs_folder = os.path.join(self.inmanta_project_dir.name, "libs")
+
+        if os.path.exists(v1_metadata_file):
+            mv1 = module.ModuleV1(project=None, path=self.get_folder_path())
+            module_name = mv1.name
+            os.symlink(self.get_folder_path(), os.path.join(libs_folder, module_name), target_is_directory=True)
```

Comment:
Open question: do we need to be compatible with any Windows machine? It's symlink support seems to have some limitations: https://docs.python.org/3/library/os.html#os.symlink.

Reply:
I forgot to address this. I could catch OSError and suggest the user activates  Developer Mode  or runs vscode as admin ?

Reply:
I just asked Bart. Apparently we are already incompatible with Windows so we can leave it as is.